.github/workflows/build_release_binaries.yml

```diff
@@ -0,0 +1,73 @@
+name: Create Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: Tag name for the release.
+        required: true
+
+jobs:
+  create_release:
+    name: Create release
+    runs-on: ubuntu-latest
+    outputs:
+      upload_url: ${{ steps.create_release.outputs.upload_url }}
+    steps:
+      - name: Create release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.event.inputs.version }}
+          release_name: Release v${{ github.event.inputs.version }}
+          draft: true
+
+  build_draft_release:
+    needs: create_release
+    strategy:
+      matrix:
+        include:
+          - runner: ubuntu-latest
+            os: linux
+
+          - runner: windows-latest
+            os: windows
+
+          - runner: macos-latest
+            os: macos
+    runs-on: ${{ matrix.runner }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_wrapper: false
+
+      - name: Install Go toolchain
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: 'src/go.mod'
+          cache-dependency-path: 'src/go.sum'
+
+      - name: Compile Test Binary
+        run: |
+          cd src/
+          go test -c -o ../bin/infra-tester
+
+      - name: Run Example Terraform Tests
+        run: |
+          cd example
+          ../bin/infra-tester -test.v
+
+      - name: Upload Release Asset
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ needs.create_release.outputs.upload_url }}
+          asset_path: ./bin/infra-tester
+          asset_name: infra_tester_${{ matrix.os }}_x86_64
```

Comment:
> It seems that the tests on Windows are not executed. I wonder whether this happens because the binary doesn't have the .exe extension and, thus, is not identified as executable by Windows.
Compiling and running the test binary in bash in windows seem to work. Let me see if setting a .exe extension will work. But this would mean separate code path for compiling and running tests for windows and (linux, mac).

Reply:
Fixed: https://github.com/schrodinger/infra-tester/actions/runs/4572311383/jobs/8071427304

It was the lack of `.exe`. I've kept the build code path to be same for all three platforms too.